homeassistant/components/wyoming/wake_word.py

```diff
@@ -0,0 +1,146 @@
+                return chunk_bytes
+
+        try:
+            async with AsyncTcpClient(self.service.host, self.service.port) as client:
+                await client.write_event(
+                    AudioStart(
+                        rate=16000,
+                        width=2,
+                        channels=1,
+                    ).event(),
+                )
+
+                # Read audio and wake events in "parallel"
+                audio_task = asyncio.create_task(next_chunk())
+                wake_task = asyncio.create_task(client.read_event())
+                pending = {audio_task, wake_task}
+
+                try:
+                    while True:
+                        done, pending = await asyncio.wait(
+                            pending, return_when=asyncio.FIRST_COMPLETED
+                        )
+
+                        if audio_task in done:
+                            # Forward audio to wake service
+                            chunk_info = audio_task.result()
+                            if chunk_info is None:
+                                break
+
+                            chunk_bytes, chunk_timestamp = chunk_info
+                            chunk = AudioChunk(
+                                rate=16000,
+                                width=2,
+                                channels=1,
+                                audio=chunk_bytes,
+                                timestamp=chunk_timestamp,
+                            )
+                            await client.write_event(chunk.event())
+
+                            # Next chunk
+                            audio_task = asyncio.create_task(next_chunk())
+                            pending.add(audio_task)
+
+                        if wake_task in done:
+                            event = wake_task.result()
+                            if event is None:
+                                _LOGGER.debug("Connection lost")
+                                break
+
+                            if Detection.is_type(event.type):
+                                # Successful detection
+                                detection = Detection.from_event(event)
+                                _LOGGER.info(detection)
+                                return wake_word.DetectionResult(
+                                    ww_id=detection.name, timestamp=detection.timestamp
+                                )
+
+                            # Next event
+                            wake_task = asyncio.create_task(client.read_event())
+                            pending.add(wake_task)
+                finally:
+                    # Clean up
+                    if audio_task in pending:
+                        # It's critical that we don't cancel the audio task or
+                        # leave it hanging. This would mess up the pipeline STT
+                        # by stopping the audio stream.
+                        await audio_task
```

Comment:
This is the chunk after the current wake word chunk. Is that one considered lost now?

Reply:
Good catch. I'll bubble this up to the pipeline.